pepdbagent/modules/annotation.py

```diff
@@ -0,0 +1,341 @@
+        """
+        if all([namespace, name, tag]):
+            found_annotations = list(
+                self._get_single_annotation(
+                    namespace=namespace,
+                    name=name,
+                    tag=tag,
+                    admin=admin,
+                )
+            )
+            return AnnotationReturnModel(
+                count=len(found_annotations),
+                limit=1,
+                offset=1,
+                result=found_annotations,
+            )
+        return AnnotationReturnModel(
+            limit=limit,
+            offset=offset,
+            count=self._count_projects(
+                namespace=namespace, search_str=query, admin=admin
+            ),
+            result=self._get_projects(
+                namespace=namespace,
+                search_str=query,
+                admin=admin,
+                offset=offset,
+                limit=limit,
+            ),
+        )
+
+    def get_by_rp(
+        self,
+        registry_paths: Union[List[str], str],
+        admin: Union[List[str], str] = None,
+    ) -> AnnotationReturnModel:
+        """
+        Get project annotations by providing registry_path or list of registry paths.
+        :param registry_paths: registry path string or list of registry paths
+        :param admin: list of namespaces where user is admin
+                :return: pydantic model: AnnotationReturnModel(
+            limit:
+            offset:
+            count:
+            result: List [ AnnotationModel(
+                        namespace:
+                        name:
+                        tag:
+                        is_private:
+                        number_of_samples:
+                        description:
+                        last_update_date:
+                        submission_date:
+                        digest:
+                    )
+                ]
+            )
+        """
+        if isinstance(registry_paths, list):
+            anno_results = []
+            for path in registry_paths:
+                try:
+                    namespace, name, tag = registry_path_converter(path)
+                except RegistryPathError as err:
+                    _LOGGER.error(str(err), registry_paths)
+                    continue
+                single_return = self._get_single_annotation(namespace, name, tag, admin)
+                if single_return:
+                    anno_results.append(single_return)
+            return_len = len(anno_results)
+            return AnnotationReturnModel(
+                count=return_len,
+                limit=return_len,
+                offset=len(registry_paths),
+                result=anno_results,
+            )
+
+        else:
+            try:
+                namespace, name, tag = registry_path_converter(registry_paths)
+            except RegistryPathError as err:
+                _LOGGER.error(str(err), registry_paths)
+                return AnnotationReturnModel()
```

Comment:
Again, here I think just throwing the exception is better. If I supply `nleroy917/basic/basic:latest` I want en error not an empty Annotation object.

Reply:
I agree